### ISSUE TITLE
`reportReturnType` false positive on protocol method with docstring

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1183]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYCwAULbaRQIYDOLUAYmGABRFglyFAJQAuWlElQAJgFNgUVtOA8WsisGFQAtAD4oLGCFFQAxFFkhwICVLsAiR0uCP7QA)

```python
from typing import Protocol


class Foo(Protocol):
    def asdf(self) -> str: # error
        """asdf"""
```
upstream issue: https://github.com/microsoft/pyright/issues/9037
